### DISCUSSION TITLE
Elastalert Backend

### DISCUSSION THREAD

**Discussion by krapgras** [SigmaHQ/pySigma#220] [Q&A] [answered]
Hello,

How difficult would it be to (re)integrate the sigma2elastalert backend?
I'm trying to revive the HELK stack and one of the steps it uses is converting the sigma rules to elastalert.
From what i can see the script [(linked below)](https://github.com/SigmaHQ/legacy-sigmatools/blob/master/contrib/sigma2elastalert.py) just uses the old sigmac es-qs and then does some text alterations via python [args.sigmac, file, "--target", "es-qs"].

Unfortunately my knowledge on the new pySigma is limited but i would imagine that the changes should be minimal to reimplement this as the es-qs backend still exists? 

[legacy-sigmatools](https://github.com/SigmaHQ/legacy-sigmatools/blob/master/contrib/sigma2elastalert.py)

**Comment by thomaspatzke** [comment, marked as answer]
Check the [YAML pipelines project](https://github.com/frack113/pySigma-yml-pipeline) from @frack113, he implemented [Elastalert support based on the Elastic backend with processing pipelines](https://github.com/frack113/pySigma-yml-pipeline/tree/main/elastalert).

**Reply by krapgras** [reply]
Thanks I totally missed this one :) That should do the trick I think!

**Reply by adilraad2001** [reply]
did convese the rules ?

**Reply by krapgras** [reply]
@adilraad2001 not sure what you mean but yes I have updated the rules a bit and it is working.

It is part of my helk fork and you can find the file [here](https://github.com/krapgras/HELK/blob/master/docker/helk-elastalert/sigmac/elastalert_any_v2.yml)

One thing I did have to do is yaml lint some of the files as not all conventions where fully correct in syntax. I didn’t try to fix them to be honest (lack of time)
